gradoop-temporal/src/main/java/org/gradoop/temporal/model/impl/TemporalGraphCollectionFactory.java

```diff
@@ -0,0 +1,231 @@
+    Map<String, DataSet<TemporalEdge>> edges) {
+
+    return new TemporalGraphCollection(layoutFactory
+      .fromIndexedDataSets(graphHeads, vertices, edges), config);
+  }
+
+  @Override
+  public TemporalGraphCollection fromCollections(Collection<TemporalGraphHead> graphHeads,
+    Collection<TemporalVertex> vertices, Collection<TemporalEdge> edges) {
+    return new TemporalGraphCollection(layoutFactory.fromCollections(graphHeads, vertices, edges), config);
+  }
+
+  @Override
+  public TemporalGraphCollection fromGraph(
+    LogicalGraphLayout<TemporalGraphHead, TemporalVertex, TemporalEdge> logicalGraphLayout) {
+    return new TemporalGraphCollection(layoutFactory.fromGraphLayout(logicalGraphLayout), config);
+  }
+
+  @Override
+  public TemporalGraphCollection fromGraphs(
+    LogicalGraphLayout<TemporalGraphHead, TemporalVertex, TemporalEdge>... logicalGraphLayout) {
+    if (logicalGraphLayout.length == 0) {
+      return createEmptyCollection();
+    } else if (logicalGraphLayout.length == 1) {
+      return fromGraph(logicalGraphLayout[0]);
+    } else {
+      return fromDataSets(
+        Arrays.stream(logicalGraphLayout).map(LogicalGraphLayout::getGraphHead).reduce(DataSet::union)
+          .get().distinct(new Id<>()),
+        Arrays.stream(logicalGraphLayout).map(LogicalGraphLayout::getVertices).reduce(DataSet::union)
+          .get().distinct(new Id<>()),
+        Arrays.stream(logicalGraphLayout).map(LogicalGraphLayout::getEdges).reduce(DataSet::union)
+          .get().distinct(new Id<>()));
+    }
+  }
+
+  @Override
+  public TemporalGraphCollection fromTransactions(DataSet<GraphTransaction> transactions) {
+    throw new UnsupportedOperationException("This operation is not (yet) supported.");
+  }
+
+  @Override
+  public TemporalGraphCollection fromTransactions(DataSet<GraphTransaction> transactions,
+    GroupReduceFunction<TemporalVertex, TemporalVertex> vertexMergeReducer,
+    GroupReduceFunction<TemporalEdge, TemporalEdge> edgeMergeReducer) {
+    throw new UnsupportedOperationException("This operation is not (yet) supported.");
+  }
+
+  @Override
+  public TemporalGraphCollection createEmptyCollection() {
+    return new TemporalGraphCollection(layoutFactory.createEmptyCollection(), config);
+  }
+
+  @Override
+  public GraphHeadFactory<TemporalGraphHead> getGraphHeadFactory() {
+    return layoutFactory.getGraphHeadFactory();
+  }
+
+  @Override
+  public VertexFactory<TemporalVertex> getVertexFactory() {
+    return layoutFactory.getVertexFactory();
+  }
+
+  @Override
+  public EdgeFactory<TemporalEdge> getEdgeFactory() {
+    return layoutFactory.getEdgeFactory();
+  }
+
+  /**
+   * Creates a {@link TemporalGraphCollection} instance by the given graph head, vertex and edge datasets.
+   *
+   * @param graphHead graph head DataSet
+   * @param vertices vertex DataSet
+   * @param edges edge DataSet
+   * @return a temporal graph collection
+   */
+  public TemporalGraphCollection fromNonTemporalDataSets(
```

Comment:
smal test should be added

Reply:
This is already tested (?)

Reply:
I'm assuming you mean the function with time extractors